metricsbp/config.go

```diff
@@ -0,0 +1,28 @@
+package metricsbp
+
+import (
+	"context"
+	"io"
+
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/tracing"
+)
+
+type Config struct {
+	Namespace           string
+	Endpoint            string
+	CounterSampleRate   *float64
+	HistogramSampleRate *float64
```

Comment:
I'd suggest that for those more complex config field names, add the yaml annotation here.

Otherwise you can only use the yaml's default conversion from field name to key name in yaml file, which is not great: https://github.com/reddit/baseplate.go/blob/63c4dda088d9f8f53af9080618bdf60135972872/server_test.go#L27